src/app/services/article/common/getAmount.ts

```diff
@@ -17,6 +17,44 @@
 
 import {ArticleModel} from 'app/db/models';
 
-export default function () : Promise <number> {
-	return ArticleModel.count({}).then(amount => amount);
+export default function (search?: string) : Promise <number> {
+	const match = {
+		feedbacks: {$not: {$eq: 0}},
+		'itemField.order.type': 1,
+		'itemField.order.item': 1,
+	};
+	if (search) {
+		match['$or'] = [
+			{'itemField.text' : new RegExp(`${search}`, 'i')},
+			{'url' : new RegExp(`^(?:http(?:s)?\\:\\/\\/)?(?:www\\.)?(?:.*\\..*\\/)${search}`, 'i')},
```

Comment:
it would be really nice to have an example of what can be parsed and what you can receive as result

Reply:
Got it